modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala

```diff
@@ -428,6 +447,128 @@ object ObservationService {
           _             <- ResultT(asterismService.setAsterism(pid, NonEmptyList.of(newOid), input.asterism))
         yield CloneIds(origOid, newOid)).value
         
+      // ***********************
+      // All about the validations
+      // ***********************
+      def cfpError(msg: String): ObservationValidation =
+        ObservationValidation.callForProposals(msg)
+
+      extension (ra: RightAscension)
+        def isInInterval(raStart: RightAscension, raEnd: RightAscension): Boolean =
+          if (raStart > raEnd) raStart <= ra || ra <= raEnd
+          else raStart <= ra && ra <= raEnd
+
+      extension (dec: Declination)
+        def isInInterval(decStart: Declination, decEnd: Declination): Boolean =
+          decStart <= dec && dec <= decEnd
+
+      extension (tsi: TimestampInterval)
+        def centerUnsafe: Timestamp =
+          if (tsi.isEmpty) tsi.start
+          else tsi.start.plusMicrosOption(tsi.boundedTimeSpan.toMicroseconds / 2).get
+
+      extension (ge: GeneratorParamsService.Error)
```

Comment:
If I'd written `GeneratorParamsService` today, I probably would have just had it return a `Result` instead of a specific `Error`.  In fact, I've been converting some of the older ones that return service-specific `Error` messages to just return `Result`.  I wonder if the `Error`s are needed after all.

Reply:
The errors are useful here for reporting to the user. But, if it returned a Result we could just extract the error messages from there. We would just have to change the ObservationValidationCodes. Maybe just `Configuration Error`.